### PR TITLE
Upload additional LineUp ranking configuration

### PR DESCRIPTION
**Pull request by thinkh**
related to Caleydo/lineup_app#47

**prerequisites**: 
 * [x] branch is up-to-date with the branch to be merged with, i.e. develop
 * [x] build is successful
 * [x] code is cleaned up and formatted 

### Summary

Users should be able to upload/import a ranking configuration (aka LineUp dump) and an additional CSV file.

#### How to use

1. Export dataset from Excel as CSV file
2. Import CSV file into LineUp to generate initial ranking configuration
   image
3. Modify ranking (e.g., move columns, change column width, mapping, remove columns)
4. Hover over download icon (top right) and select _Download JSON dump_
   image
5. Modify the downloaded ranking configuration in an external editor
   image
6. Navigate to the LineUp start page (i.e., dataset selector)
7. Select the ranking configuration (JSON file) and the dataset (CSV file)
   image
8. Work with the loaded ranking
   image

#### Known issues

* **Missing columns after import**
  Downloading the ranking configuration might export the CSV columns in lower-case (e.g., `"desc": "categorical@current_league` for the column name `Current League`). On the next import LineUp cannot match the ranking configuration with the CSV columns. Hence, edit the configuration and change the column names to upper-case (e.g., `"desc": "categorical@Current_League",`). Replace spaces with underscores `_`.

### CODE REVIEW

**Comment by thinkh** [comment]
It seems that CircleCI has some troubles with the public GitHub key and terminates the build.

```
Cloning into '.'...
Warning: Permanently added the RSA host key for IP address '192.30.253.113' to the list of known hosts.

Permission denied (publickey).

fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
Exited with code 128
```

**Comment by sgratzl** [comment]
why do I need to upload the raw CSV file again? the dump automatically contains the raw data. 

e.g. when I upload the following file, and create a dump out of it, it will produce a dump with the `rawData` attribute.

[test.csv.txt](https://github.com/lineupjs/lineup_app/files/3078072/test.csv.txt)
[test.json.txt](https://github.com/lineupjs/lineup_app/files/3078073/test.json.txt)

File: src/data/index.ts [hunk 1]
```diff
@@ -59,9 +59,30 @@ export function fromFile(file: File): Promise<IDataset> {
       return loader.loadFile(file).then(complete);
     }
   }
-  return Promise.reject(`unknown file type: ${file.name}`);
+  return Promise.reject(`Unknown file type: ${file.name}`);
 }
 
+export function checkFiles(files: File[]): Promise<IDataset> {
+  const jsonFile = files.filter((f) => JSON_LOADER.supports(f));
```
Comment (sgratzl) [inline src/data/index.ts:66]: why not using the `find` function instead?

File: src/index.ts [hunk 2]
```diff
@@ -202,15 +202,45 @@ function recreateCard(dataset: IDataset) {
...
-    shared.datasets.push(r);
-    addToCarousel(r);
-    refreshCarousel();
-    return storeDataset(r).then(() => r);
-  });
-  build(f);
+  shared.datasets.push(dataset);
+  addToCarousel(dataset);
+  refreshCarousel();
+  storeDataset(dataset);
```
Comment (sgratzl) [inline src/index.ts:240]: why did you remove to wait for the `storeDataset` to finish?
